pkgs/applications/science/electronics/kicad/kicad-ngspice.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/eeschema/sim/ngspice.cpp b/eeschema/sim/ngspice.cpp
+index 31885c2a6..fb5f19b8e 100644
+--- a/eeschema/sim/ngspice.cpp
++++ b/eeschema/sim/ngspice.cpp
+@@ -287,7 +287,7 @@ void NGSPICE::init_dll()
+         m_dll.Unload();
+ 
+ // Extra effort to find libngspice
+-#if defined(__WINDOWS__) || (__WXMAC__)
++#if defined(__WINDOWS__) || (__WXMAC__) || (__LINUX__)
+ #ifdef __WINDOWS__
+     wxFileName dllFile( "", NGSPICE_DLL_FILE );
+     const vector<string> dllPaths = { "", "/mingw64/bin", "/mingw32/bin" };
+@@ -303,6 +303,10 @@ void NGSPICE::init_dll()
+         wxFileName( stdPaths.GetExecutablePath() ).GetPath() + "/../../../../../Contents/PlugIns/sim"
+     };
+ #endif /* __WXMAC__ */
++#ifdef __LINUX__
++    wxFileName dllFile( "", "libngspice.so" );
++    const vector<string> dllPaths = { "", "__NIX_STORE_LIBNGSPICE_PATH__" };
```

Comment:
I have a suspicion this patch does nothing for you. You did not include a bit from https://github.com/NixOS/nixpkgs/issues/63798#issuecomment-539872139 that would actually substitute the __NIX_STORE_LIBNGSPICE_PATH__ with some actual path.

Reply:
That would mean that my previous fix is already complete, which does not explain why kicad reliably crashed for me when I tried it without that patch. I'm just going to revert ab0280e40925177301bbfad16a27da86501a9671 in my local copy and try again to see if it still crashes for me when running a simulation.

Reply:
... and it's working without ab0280e...
So I'm going to try again figuring out exactly what is still broken and what's working tomorrow.

Reply:
I removed this commit.